pkg/query/querier.go

```diff
@@ -185,6 +187,16 @@ func (q *querier) Select(params *storage.SelectParams, ms ...*labels.Matcher) (s
 	}
 	queryAggrs, resAggr := aggrsFromFunc(params.Func)
 
+	matchers := make([]string, len(ms))
```

Comment:
I think this can work for now, but what about doing it properly on GRPC StoreAPI level? Potentially as interceptor even? :thinking:

Reply:
I wonder if there is some existing interceptor for it TBH..

Reply:
Yeah I was thinking about this. For now I want it to concentrate on the most essential things we need. I'll keep thinking about this though.

Reply:
There is none: Added issue: https://github.com/grpc-ecosystem/go-grpc-middleware/issues/273